src/Processors/Transforms/buildPushingToViewsChain.cpp

```diff
@@ -684,10 +686,11 @@ void PushingToWindowViewSink::consume(Chunk chunk)
 }
 
 
-FinalizingViewsTransform::FinalizingViewsTransform(std::vector<Block> headers, ViewsDataPtr data)
+FinalizingViewsTransform::FinalizingViewsTransform(std::vector<Block> headers, ViewsDataPtr data, bool materialized_views_ignore_errors_)
     : IProcessor(initPorts(std::move(headers)), {Block()})
     , output(outputs.front())
     , views_data(std::move(data))
+    , materialized_views_ignore_errors(materialized_views_ignore_errors_)
```

Comment:
I think we can get away with not adding a new parameter to the class and instead read it from data:

```c++
bool materialized_views_ignore_errors = views_data->context->getSettingsRef().materialized_views_ignore_errors;
```

The less parameters the better IMO

Reply:
Yes, make sense. Thanks.